### DISCUSSION TITLE
OCI runtime create failed: container_linux.go:345

### DISCUSSION THREAD

**Discussion by sefasenturk95** [community/community#26193] [Actions] [answered]
<p>Hi,</p>
<p>Currently we are migrating from GitLab To GitHub and we decided to use GitHub Actions. The building of the go application and pushing it to GKE works perfectly. But when GKE tries to spin up the new version of the pod it gives back an error: </p>
<pre><code>'OCI runtime create failed: container_linux.go:345: starting container
          process caused "exec: \"/socket-server\": permission denied": unknown'
</code></pre>
<p>My workflow file looks like this:</p>
<pre><code>name: Build and deploy
on:
  - push
  - pull_request

jobs:
  build:
    name: Build
    runs-on: ubuntu-latest
    steps:
      - uses: fusion-engineering/setup-git-credentials@v2
        with:
          credentials: https://${{ secrets.MACHINE_ACCOUNT_ACCESS_TOKEN }}:x-oauth-basic@github.com/

      - name: Setup environment
        shell: bash
        run: |
          echo "::set-env name=GOPATH::${{ github.workspace }}/go"
          echo "::add-path::${{ github.workspace }}/go/bin"

      - name: Install Go
        uses: actions/setup-go@v1
        with:
          go-version: 1.12.4

      - name: Checkout code
        uses: actions/checkout@v2
        with:
          path: go/src/github.com/${{ github.repository }}

      - name: Prepare environment
        run: |
          cd $GOPATH/src/github.com/${{ github.repository }}
          make prepare

      - name: Format code
        run: |
          cd $GOPATH/src/github.com/${{ github.repository }}
          make fmt &amp;&amp; git diff --exit-code

      - name: Lint code
        run: |
          cd $GOPATH/src/github.com/${{ github.repository }}
          make lint

      - name: Vet code
        run: |
          cd $GOPATH/src/github.com/${{ github.repository }}
          make vet

      - name: Test code
        run: |
          cd $GOPATH/src/github.com/${{ github.repository }}
          make cover

      - name: Build code
        run: |
          cd $GOPATH/src/github.com/${{ github.repository }}
          make build

      - name: Upload artifact
        uses: actions/upload-artifact@v1
        with:
          name: socket-server
          path: go/src/github.com/${{ github.repository }}/socket-server

  deploy:
    name: Deploy
    runs-on: ubuntu-latest
    needs: [build]
    if: contains(github.ref, 'refs/tags')

    steps:
      - name: Set release version
        run: echo ::set-env name=CI_COMMIT_TAG::${GITHUB_REF/refs\/tags\//}

      - name: Checkout code
        uses: actions/checkout@v2

      - name: Get artifact from build step
        uses: actions/download-artifact@v1
        with:
          name: socket-server

      - name: Set ci auth
        run: echo ::set-env name=CI_AUTH::$(cat ci_auth.json | base64)

      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
        with:
          service_account_key: ${{ env.CI_AUTH }}

      - name: Configure gcloud docker authentication
        run: |
          gcloud config set project foo
          gcloud auth configure-docker

      - name: Build, push and deploy container
        run: |
          bash deploy.sh
</code></pre>
<p>The deploy.sh looks like this:</p>
<pre><code>#!/bin/bash

if [[-z "${CI_COMMIT_TAG}"]]; then
    echo "CI_COMMIT_TAG is empty, this stage should not run"
    exit 0
fi

export ENV="stage"

if [["$CI_COMMIT_TAG" != "${CI_COMMIT_TAG%-release}"]]; then
    export ENV="prod"
fi

echo "Current environment: $ENV"

make deploy
</code></pre>
<p>And the deploy step in the makefile looks like this:</p>
<pre><code>deploy:
	( echo "cat &lt;&lt;EOF" ; cat k8s.yml.template; ) | sh &gt; k8s-${ENV}.yml
	docker build --no-cache \
	--build-arg RELEASE=${CI_COMMIT_TAG} \
	--build-arg ENV=${ENV} \
	-t gcr.io/foo/socket-server:${CI_COMMIT_TAG} .
	docker push gcr.io/foo/socket-server:${CI_COMMIT_TAG}
	gcloud container clusters get-credentials api-${ENV} --zone=europe-west1-b
	kubectl apply -f k8s-${ENV}.yml
</code></pre>
<p>And the Dockerfile looks like this:</p>
<pre><code>FROM alpine:latest as certs
RUN apk --update add ca-certificates

FROM scratch
COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

ARG RELEASE
ARG ENV

ADD ./socket-server /socket-server
ADD ./config.yml /config.yml
ADD ./dbconfig.yml /dbconfig.yml
ADD ./migrations /migrations

ENV SOCKET_SERVER_SENTRY_DSN https://foo@sentry.io/bar
ENV SOCKET_SERVER_SENTRY_RELEASE $RELEASE
ENV SOCKET_SERVER_SENTRY_ENVIRONMENT $ENV

CMD ["/socket-server", "--port", "9345", "--host", ""]
</code></pre>
<p>This whole deployment was working in the GitLab CI/CD and I am pretty sure that I am missing something here but I can’t see it. I have already tried chmod +x socket-server on the pipeline and in the Dockerfile but that did not work. I’ve also opened a Stackoverflow question so feel free to comment over there or check it out: <a href="https://stackoverflow.com/questions/60339290/oci-runtime-create-failed-container-linux-go345/60341155" rel="nofollow noopener">https://stackoverflow.com/questions/60339290/oci-runtime-create-failed-container-linux-go345/60341155</a></p>
<p>Ideas anyone? Thanks in advance!</p>

**Comment by Yanjingzhu** [comment]
<p>According to your dockerfile and workflow yml, after download artifacts step, a folder named  <strong>socket-server </strong> will show on your github workspace. You could not make a folder executable. </p>
<p>I am not sure which file you want to run under socket-server, if it called test.sh , you could try to modify your dockerfile with </p>
<pre><code>RUN chmod +x /socket-server/test.sh
CMD ["/socket-server/test.sh", "--port", "9345", "--host", ""]
</code></pre>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/sef1995">@sef1995</a>  To make myself clearer, now I add some screenshots to explain it. </p>
<p>Accoding to your workflow file, you upload artifact named  <strong>socket-server</strong> in job <strong>build</strong>. Then download it in job <strong>Deploy</strong>. In your working directry there will be a folder named socket-server . You could not make a <strong>folder</strong> executable using chmod +x socket-server . </p>
<p>In my example the artifact is named socket-server1.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4b1aecbbbbbbd90bafb48f473b33a5849efd73dd.png" data-download-href="/uploads/short-url/aIpuLFyE8HMkrByChZ6lHI0bLrL.png?dl=1" title="Upload artifact.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181096014-b73f1aec-c9a7-42b9-b150-edb0e2f5f9ec.png" alt="Upload artifact.png" data-base62-sha1="aIpuLFyE8HMkrByChZ6lHI0bLrL" width="690" height="283" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4b1aecbbbbbbd90bafb48f473b33a5849efd73dd_2_690x283.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/4b1aecbbbbbbd90bafb48f473b33a5849efd73dd_2_1035x424.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/4b1aecbbbbbbd90bafb48f473b33a5849efd73dd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4b1aecbbbbbbd90bafb48f473b33a5849efd73dd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Upload artifact.png</span><span class="informations">1232×506 39.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So I recommand you add the sub file name in chmod +x command to <strong>Dockerfile</strong>. For example:  </p>
<pre><code>RUN chmod +x /socket-server/test.sh
CMD ["/socket-server/test.sh", "--port", "9345", "--host", ""] 
</code></pre>
<p> Hope this could help. </p>
